typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -75,6 +73,44 @@ export class LiquidityLayerApp<
     return response.result.map((_: any) => _.transactionHash).flat();
   }
 
+  async fetchPortalBridgeTransactions(chain: Chain): Promise<string[]> {
+    const cc = this.multiProvider.getChainConnection(chain);
+    const params = new URLSearchParams({
+      module: 'logs',
+      action: 'getLogs',
+      address: this.getContracts(chain).portalAdapter!.address,
+      topic0: PortalBridgedTokenTopic,
+    });
+    const req = await fetch(`${cc.getApiUrl()}?${params}`);
+    const response = await req.json();
+
+    return response.result.map((_: any) => _.transactionHash).flat();
+  }
+
+  async parsePortalMessages(
+    chain: Chain,
+    txHash: string,
+  ): Promise<PortalBridgeMessage<Chain>[]> {
+    const connection = this.multiProvider.getChainConnection(chain);
+    const receipt = await connection.provider.getTransactionReceipt(txHash);
+    const matchingLogs = receipt.logs
+      .map((_) => {
+        try {
+          return [PortalAdapterInterface.parseLog(_)];
+        } catch {
+          return [];
+        }
+      })
+      .flat();
+    if (matchingLogs.length == 0) return [];
+
+    const event = matchingLogs.find((_) => _!.name === 'BridgedToken')!;
+    const portalSequence = event.args.portalSequence.toNumber();
+    const nonce = event.args.nonce.toNumber();
+    const destination = DomainIdToChainName[event.args.destination];
```

Comment:
`parseLog` somehow does not return a event specific type, instead it does a generic `LogDescription` which means that all event args are untyped

Reply:
Untyped at the typescript layer but do have a time at runtime, and I've found that ethers turns all number-ish types in BigNumbers

Reply:
Ah sorry I misinterpreted. So I think it turns number-ish types to BigNumber when the number itself is possibly a big number. Since domains are u32, this can be represented as anumber and i guess this is indeed converted as such